r/vignettes/install.Rmd

```diff
@@ -7,49 +7,215 @@ vignette: >
+click on 'binary', and select your preferred distribution.
+
+Similarly, if you use `conda` to manage your R environment, you can get the 
+latest official release of the R package including libarrow via:
+
+```shell
+conda install -c conda-forge --strict-channel-priority r-arrow
+```
+
+### Method 1b - R source package with libarrow binary
+
+```{r, echo=FALSE, out.width="50%"}
+knitr::include_graphics("./r_source_libarrow_binary.png")
+```
+
+Another way of achieving faster installation with all key features enabled is to use our self-hosted libarrow binaries.  You can do this by setting the `NOT_CRAN` environment variable before you call `install.packages()`:
+
+```{r, eval = FALSE}
+Sys.setenv("NOT_CRAN" = TRUE)
+install.packages("arrow")
+```
+
+This installs the source version of the R package, but during the installation process will check for compatible libarrow binaries that we host and use those if available.  If no binary is available or can't be found, then this option falls back onto method 2 below, but results in a more fully-featured build than default.
+
+## Method 2 - Installing an R source package and building libarrow from source
+
+```{r, echo=FALSE, out.width="50%"}
+knitr::include_graphics("./r_source_libarrow_source.png")
+```
 
 Generally compiling and installing R packages with C++ dependencies, requires 
 either installing system packages, which you may not have privileges to do, or 
 building the C++ dependencies separately, which introduces all sorts of 
-additional ways for things to go wrong. 
+additional ways for things to go wrong, which is why we recommend method 1 above.
 
-Note also that if you use `conda` to manage your R environment, this document does not apply.
-You can `conda install -c conda-forge --strict-channel-priority r-arrow` and you'll get the latest official
-release of the R package along with any C++ dependencies.
+However, if you wish to fine-tune or customise your Linux installation, the 
+instructions in this section explain how to do that.
 
-> Having trouble installing `arrow`? See the "Troubleshooting" section below.
+### Basic configuration for building from source with fully featured installation
 
-# Installation basics
+If you wish to install libarrow from source instead of looking for pre-compiled 
+binaries, you can set the  `LIBARROW_BINARY` variable.
 
-Install the latest release of `arrow` from CRAN with
+```{r, eval = FALSE}
+Sys.setenv("LIBARROW_BINARY" = FALSE)
+```
 
-```r
-Sys.setenv(NOT_CRAN = TRUE)
+By default, this is set to `TRUE`, and so libarrow will only be built from 
+source if this environment variable is set to `FALSE` or no compatible binary 
+for your OS can be found.
+
+When compiling libarrow from source, you have the power to really fine-tune 
+which features to install.  You can set the environment variable 
+`LIBARROW_MINIMAL` to enable a more full-featured build including S3 support 
```

Comment:
I think we should have something like `LIBARROW_MINIMAL=FALSE` here since it's a little bit odd with this one variable that `FALSE` is the one that someone wants and not `TRUE`

Reply:
Done!